src/main/java/com/browserstack/automate/ci/jenkins/pipeline/BrowserStackPipelineStepExecution.java

```diff
@@ -87,7 +88,8 @@ protected Void run() throws Exception {
                         localConfig, browserStackLocal);
 
         EnvVars overrides = run.getEnvironment(taskListener);
-        HashMap<String, String> overridesMap = new HashMap<String, String>(overrides);
+        HashMap<String, String> overridesMap = new HashMap<String, String>();
+        overridesMap.put(Constants.JENKINS_BUILD_TAG, overrides.get(Constants.JENKINS_BUILD_TAG));
```

Comment:
Here you are removing all the existing keys of overrides except JENKINS_BUILD_TAG, won't this remove the app id or any other credentials present in that map?

Reply:
App ID is not present in `run.getEnvironment(taskListener)` anyway. It only includes those env vars which are added by the Jenkins, i.e. BUILD_TAG etc.

And the reason for adding `BUILD_TAG` is because its value is used to decide the value for `BROWSERSTACK_BUILD_NAME` in `BrowserStackBuildWrapperOperations`: https://github.com/jenkinsci/browserstack-integration-plugin/blob/f5a3057d8a6f667fa7328e02cec72382b1ca2525/src/main/java/com/browserstack/automate/ci/common/BrowserStackBuildWrapperOperations.java#L130

Reply:
Regarding Credentials, they are added in the line 93 (just below it) : https://github.com/jenkinsci/browserstack-integration-plugin/pull/24/files/cd7424dbc5ed0ddf2dce95442f0288c5015d8965#diff-849941d08d01b488b765a66e356cc855652c5d1d949d36037e460a74ce16cbb1R93

Reply:
but, removing other env vars from overrides won't result into any issues?

Reply:
We hadn't received any issues around that earlier. But, post adding `run.getEnvironment(taskListener)` env vars led to behaviour change, for which we received a ticket. Thus, changing the behaviour back to the way it was.

Reply:
But yes... we aren't sure of the exhaustive consequences.

Reply:
Just to clarify:
Even if we don't add `run.getEnvironment(taskListener)` env vars in out `overrides`, we do have access to the Jenkins ENV vars. i.e. The following pipeline file logs `BUILD_URL` in both the blocks:

```
pipeline {
    agent any

    stages {
        stage('Hello') {
            steps {
                browserstack('02da61c7-5966-49d7-8ebb-05d3a087814e') {
                    echo "HERE..."
                    echo "BUILD NAME: ${BROWSERSTACK_BUILD_NAME}"
                    echo "BUILD URL: ${BUILD_URL}"
                }
                
                browserstackAppUploader('/Users/chaugs_rohan/Downloads/Instagram_v157.0.0.37.120_apkpure.com.apk') {
                    // some block
                    echo "IN APP UPLOAD..."
                    echo "BROWSERSTACK_APP_ID: ${BROWSERSTACK_APP_ID}"
                    echo "BUILD URL: ${BUILD_URL}"
                }
            }
        }
    }
    post {
        always {
          echo "calling the publisher pipeline..."
          browserStackReportPublisher "automate"
        }
    }
}

```

Result:
![image](https://user-images.githubusercontent.com/22256912/96453357-f81bba00-1237-11eb-803c-9e2fd90fa044.png)

Reply:
Nice